pontoon/base/views.py

```diff
@@ -306,21 +306,46 @@ def get_translations_info(translations):
         return count, translated_resources, changed_entities
 
     if action == 'approve':
-        translations = translations.filter(approved=False)
-        changed_translation_pks = list(translations.values_list('pk', flat=True))
+        approved = translations.filter(approved=False)
+        changed_translation_pks = list(approved.values_list('pk', flat=True))
         if changed_translation_pks:
-            latest_translation_pk = translations.last().pk
-        count, translated_resources, changed_entities = get_translations_info(translations)
-        translations.update(
+            latest_translation_pk = approved.last().pk
+        count, translated_resources, changed_entities = get_translations_info(approved)
+        approved.update(
             approved=True,
             approved_user=request.user,
-            approved_date=timezone.now()
+            approved_date=timezone.now(),
+            rejected=False,
+            rejected_user=None,
+            rejected_date=None,
+            fuzzy=False,
         )
 
-    elif action == 'delete':
-        count, translated_resources, changed_entities = get_translations_info(translations)
-        TranslationMemoryEntry.objects.filter(translation__in=translations).delete()
-        translations.delete()
+        # Reject all other non-rejected translations.
+        suggestions = translations.filter(approved=False, rejected=False)
+        suggestions.update(
+            rejected=True,
+            rejected_user=request.user,
+            rejected_date=timezone.now(),
+            fuzzy=False,
+        )
```

Comment:
You need to reject non-rejected suggestions to strings for which you just approved active translations for. (This sounds funny, but it's true).

Instead you're currently rejecting active non-rejected suggestions (which don't even exist, because you just approved a superset of it - `approved`).